packages/iTwinUI-react/src/core/Dialog/DialogMain.tsx

```diff
@@ -109,6 +113,16 @@ export const DialogMain = React.forwardRef<HTMLDivElement, DialogMainProps>(
       onKeyDown?.(event);
     };
 
+    const { onPointerDown, transform } = useDragAndDrop(dialogRef);
+    const handlePointerDown = React.useCallback(
```

Comment:
Not sure I understand what's going on here. You grab the `onPointerDown` callback from context, wrap it in `handlePointerDown` and pass it back into the provider?

Reply:
No, I take pass main dialog ref and wrapper bounding rect to the `useDragAndDrop` and take `onPointerDown` from it, then pass it down the context.

Reply:
should we just pass the ref down in context then? because this `handlePointerDown` is not used in this component

Reply:
wait i see we are using the `transform` in here. hmm, we could do `dialogMainRef.current.style.transform = transform` in title bar, but maybe that's confusing.

Reply:
We can't just apply `transform` to the html element because if dialog is hidden by setting `isOpen={false}` then element disappears and `transform` is lost.